src/value.rs

```diff
@@ -2,8 +2,80 @@ use u256::U256;
 
 /// In the zkEVM, all data in the stack and on registers is tagged to determine
 /// whether they are a pointer or not.
-#[derive(Debug, Clone)]
+#[derive(Debug, Clone, Copy)]
 pub struct TaggedValue {
     pub value: U256,
     pub is_pointer: bool,
 }
+#[derive(Debug, Default)]
+pub struct FatPointer {
+    pub offset: u32,
+    pub page: u32,
+    pub start: u32,
+    pub len: u32,
+}
+
+impl Default for TaggedValue {
+    fn default() -> Self {
+        Self {
+            value: U256::zero(),
+            is_pointer: false,
+        }
+    }
+}
+
+impl TaggedValue {
+    pub fn new(value: U256, is_pointer: bool) -> Self {
```

Comment:
I think this function is redundant since we already have the other 2 below.

Reply:
Done